src/scaler/mod.rs

```diff
@@ -39,18 +60,363 @@ pub trait Scaler {
+                trace!("Scaler generated commands, notifying other scalers to register expected events");
+                let data = serde_json::to_vec(&Notifications::RegisterExpectedEvents {
+                    name: model_name.to_owned(),
+                    scaler_id: self.scaler.id().to_owned(),
+                    triggering_event: Some(event.to_owned().try_into()?),
+                })?;
+
+                self.notifier
+                    .publish(data, Some(&self.notify_subject))
+                    .await?;
+            }
+            self.add_events(expected_events, false).await;
+            commands
+        };
+
+        Ok(commands)
+    }
+
+    #[instrument(level = "trace", skip_all, fields(scaler_id = %self.id()))]
+    async fn reconcile_internal(&self) -> Result<Vec<Command>> {
+        // If we're already in backoff, return an empty list
+        let current_event_count = self.event_count().await;
+        if current_event_count > 0 {
+            trace!(%current_event_count, "Scaler is backing off, not reconciling");
+            return Ok(Vec::with_capacity(0));
+        }
+        match self.scaler.reconcile().await {
+            // "Back off" scaler with expected corresponding events if the scaler generated commands
+            Ok(commands) if !commands.is_empty() => {
+                trace!("Reconcile generated commands, notifying other scalers to register expected events");
+                let data = serde_json::to_vec(&Notifications::RegisterExpectedEvents {
+                    name: self.model_name.to_owned(),
+                    scaler_id: self.scaler.id().to_owned(),
+                    triggering_event: None,
+                })?;
+                self.notifier
+                    .publish(data, Some(&self.notify_subject))
+                    .await?;
+                self.add_events(
+                    commands
+                        .iter()
+                        .filter_map(|command| command.corresponding_event(&self.model_name)),
+                    true,
+                )
+                .await;
+                Ok(commands)
+            }
+            Ok(commands) => {
+                trace!("Reconcile generated no commands, no need to register expected events");
+                Ok(commands)
+            }
+            Err(e) => Err(e),
+        }
+    }
+
+    /// Sets a timed cleanup task to clear the expected events list after a timeout
+    async fn set_timed_cleanup(&self) {
+        let mut event_cleaner = self.event_cleaner.lock().await;
+        // Clear any existing handle
+        if let Some(handle) = event_cleaner.take() {
+            handle.abort();
+        }
+        let expected_events = self.expected_events.clone();
+        let timeout = self.cleanup_timeout;
+
+        *event_cleaner = Some(tokio::spawn(async move {
+            tokio::time::sleep(timeout).await;
+            // We don't use the method to clear here because of borrowing semantics.
```

Comment:
I don't understand this comment, since we are calling `clear` on the next line

Reply:
I think it was left over from something before the refactor